src/NuGet.Clients/VsConsole/PowerShellHost/PowerShellHost.cs

```diff
@@ -585,6 +600,19 @@ public bool Execute(IConsole console, string command, params object[] inputs)
                 throw new ArgumentNullException(nameof(command));
             }
 
+            var newSolutionDirectory = GetSolutionDirectory();
+
+            // check if we have already run init scripts for this solution, if not then run init scripts before 
+            // executing any script on this solution
+            if (string.IsNullOrEmpty(_currentSolutionDirectory) ||
+                !(string.IsNullOrEmpty(newSolutionDirectory) || _currentSolutionDirectory.Equals(newSolutionDirectory, StringComparison.OrdinalIgnoreCase)))
+            {
+                NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
+                {
+                    await ExecuteInitScriptsAsync();
```

Comment:
It would help to mention that install.ps1/uninstall.ps1 scripts could depend on init.ps1 scripts, and that is why we need to run this first.